src/CommandingOfficers/Tech.java

```diff
@@ -223,23 +225,41 @@ public GameEventQueue getEvents(MapMaster gameMap)
       Set<XYCoord> dropLocs = new HashSet<XYCoord>();
       for( int i = 0; i < numDrops; ++i )
       {
-        TeleportEvent techDrop = generateDropEvent(gameMap, dropLocs, log);
+        TeleportAction techDrop = generateDropEvent(gameMap, dropLocs, log);
         if( null != techDrop )
         {
           // Store the new event.
-          dropLocs.add(techDrop.getEndPoint());
-          abilityEvents.add(techDrop);
+          dropLocs.add(techDrop.getMoveLocation());
+          abilityEvents.addAll(techDrop.getEvents(gameMap));
         }
         else
         {
           System.out.println("[TechdropAbility.getEvents] WARNING! Could not find drop location.");
         }
       }
 
+      // Figure out if this caused the enemy's defeat. The TeleportActions can't track
+      // this because multiple units may be smashed at once.
+      Map<Commander, Integer> smashes = new HashMap<Commander, Integer>();
+      for( XYCoord xyc : dropLocs )
+      {
+        Unit r = gameMap.getLocation(xyc).getResident();
+        if( null != r )
+        {
+          Integer oldVal = smashes.putIfAbsent(r.CO, 1);
+          smashes.put(r.CO, (oldVal == null) ? 1 : (oldVal+1));
+        }
+      }
+      for( Commander co : smashes.keySet() )
+      {
+        if( co.units.size() <= smashes.get(co) )
+          abilityEvents.add(new CommanderDefeatEvent(co));
```

Comment:
So if you smush his last existing unit with a power, you get two defeat events?

Reply:
Yes. Realized that after I pushed it up. Not that two defeat events should break anything, but I should probs fix that.